micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -155,11 +155,11 @@ void checkAndBindMetrics(MeterRegistry registry) {
                 for (Meter other : registry.find(meterName).meters()) {
                     List<Tag> tags = other.getId().getTags();
                     // Only consider meters from the same client before filtering
-                    if (differentClient(tags)) break;
+                    if (differentClient(tags, metric.metricName().tags().get(CLIENT_ID_TAG_NAME))) continue;
```

Comment:
Previously this always used the same client ID for comparison, which wasn't correct. It also would break the loop if the client was different, but the meter with the same client could be returned after.